src/graph/entities/edge.c

```diff
@@ -49,6 +50,28 @@ Node* Edge_GetDestNode(Edge *e) {
 	return e->dest;
 }
 
+GrB_Matrix Edge_GetMatrix(Edge *e) {
+    assert(e);
+
+    // Retrieve matrix from graph if edge matrix isn't set.
+    if(!e->mat) {
+        GraphContext *gc = GraphContext_GetFromTLS();
+        Graph *g = gc->g;
+
+        // Get relation matrix.
+        if(e->relationId == GRAPH_NO_RELATION) {
+            e->mat = Graph_GetAdjacencyMatrix(g);
```

Comment:
`Graph_GetRelationMatrix` works for this case, so we can combine this and the `else` case if desired. (Not necessary if you think this is more readable.)

Reply:
👍